lib/gemonames.rb

```diff
@@ -7,15 +7,34 @@ module Gemonames
   module_function
   BASE_API_URL = "http://api.geonames.org"
 
-  def client(username:, connection: nil)
-    connection ||= build_connection(username: username)
+  def client(username: nil, connection: nil, token: nil)
```

Comment:
Avoid default parameters if possible? Maybe we need two methods? Maybe two strategies? Maybe we should drop authenticating by username?